### PR TITLE
[NEXT RELEASE] #1127 Adds documentation for Instant matchers

### PR DESCRIPTION
**Pull request by ashishkujoy**
This PR adds documentation for Instant Matchers add under [PR](https://github.com/kotlintest/kotlintest/pull/1128).

### CODE REVIEW

File: doc/matchers.md [hunk 1]
```diff
@@ -292,4 +292,13 @@ For the extension function style, each function has an equivalent negated versio
...
+| `ktype.shouldBeOfType<T>()` | Asserts that the KType has the type T |
+
+| Instant ||
+| -------- | ---- |
+| `instant.shouldBeAfter(anotherInstant)` | Asserts that the instant is after anotherInstant |
+| `instant.shouldNotBeAfter(anotherInstant)` | Asserts that the instant is not after anotherInstant which means the instants are either equal or instant is before anotherInstant |
+| `instant.shouldBeBefore(anotherInstant)` | Asserts that the instant is before anotherInstant |
+| `instant.shouldNotBeBefore(anotherInstant)` | Asserts that the instant is not before anotherInstant which means the instants are either equal or instant is after anotherInstant |
+| `instant.shouldBeBetween(fromInstant, toInstant)` | Asserts that the instant is between fromInstant and toInstant |
+| `instant.shouldNotBeBefore(fromInstant, toInstant)` | Asserts that the instant is not between fromInstant and toInstant |
```
Comment (LeoColman) [inline doc/matchers.md]: We don't document `shouldNot`, as it's always expected to have a `not` matchers. Could you remove them, please?

Comment (ashishkujoy) [reply doc/matchers.md]: sure

**Comment by LeoColman** [comment]
I'm wondering if we should add this as `Instant` matchers or if we should consider them as part of the other `Date` matchers

**Comment by ashishkujoy** [comment]
> I'm wondering if we should add this as `Instant` matchers or if we should consider them as part of the other `Date` matchers

Correct me if i am wrong, did you mean keeping them in file where we have Date matchers?

**Comment by LeoColman** [comment]
No. I meant keeping the documentation reference close to where `Date` documentation is, and thus this commit might be unnecessary.
There's already docs for `shouldBeBetween` for `LocalDates`, for example, and it's somewhere in the matchers documentation already

**Comment by sksamuel** [comment]
@ashishkujoy can you fix the conflict and merge whenever you want.

**Comment by ashishkujoy** [comment]
> @ashishkujoy can you fix the conflict and merge whenever you want.

I cannot merge as I don't have write access.

**Comment by sksamuel** [comment]
Try now @ashishkujoy

**Comment by ashishkujoy** [comment]
@sksamuel changes are now merged in master. Closing this pull request now.

**Pull request closed without merging**